File: 509. Fibonacci Number/solution.js
/**
 * @param {number} N
 * @return {number}
 */
var fib = function(N) {
  if (N <= 1) return N;

  let first = 0;
  let second = 1;
  let result = 0;
  for (let i = 2; i <= N; i++) {
    result = first + second;
    first = second;
    second = result;
  }
  return result;
};

File: 509. Fibonacci Number/solution.go
func fib(N int) int {
	if N == 0 || N == 1 {
		return N
	}
	dp := make([]int, N+1)
	dp[0] = 0
	dp[1] = 1
	for i := 2; i <= N; i++ {
		dp[i] = dp[i-1] + dp[i-2]
	}
	return dp[N]
}